src/main/java/org/elasticsearch/plugin/WordDelimiterPlugin.java

```diff
@@ -1,37 +1,53 @@
 package org.elasticsearch.plugin;
 
-import org.elasticsearch.plugins.AbstractPlugin;
-import org.elasticsearch.index.analysis.AnalysisModule;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.index.analysis.TokenFilterFactory;
+import org.elasticsearch.indices.analysis.AnalysisModule;
+import org.elasticsearch.plugins.AnalysisPlugin;
 import org.elasticsearch.common.component.LifecycleComponent;
 import org.elasticsearch.module.WordDelimiterService;
-import org.skroutz.elasticsearch.index.analysis.WordDelimiterBinderProcessor;
+import org.elasticsearch.plugins.Plugin;
 
-import java.util.Collection;
-import static org.elasticsearch.common.collect.Lists.newArrayList;
+import java.util.*;
+import java.util.function.Function;
 
-public class WordDelimiterPlugin extends AbstractPlugin {
-  @SuppressWarnings("rawtypes")
-  private final Collection<Class<? extends LifecycleComponent>> services = newArrayList();
+import org.skroutz.elasticsearch.index.analysis.WordDelimiterTokenFilterFactory;
 
-  public WordDelimiterPlugin() {
-    services.add(WordDelimiterService.class);
-  }
-
-  public String name() {
-    return "dynamic-word-delimiter";
-  }
+public class WordDelimiterPlugin extends Plugin implements AnalysisPlugin {
 
-  public String description() {
-    return "Dynamic word delimiter customized for the needs of www.skroutz.gr";
+  @Override
+  public Collection<Class<? extends LifecycleComponent>> getGuiceServiceClasses() {
+    return Collections.singleton(WordDelimiterService.class);
   }
 
-  public void onModule(AnalysisModule module) {
-    module.addProcessor(new WordDelimiterBinderProcessor());
+  @Override
+  public Map<String, AnalysisModule.AnalysisProvider<TokenFilterFactory>> getTokenFilters() {
+    return Collections.singletonMap("dynamic_word_delimiter",
+            WordDelimiterTokenFilterFactory::new);
   }
 
-  @SuppressWarnings("rawtypes")
   @Override
-  public Collection<Class<? extends LifecycleComponent>> services() {
-    return services;
+  public List<Setting<?>> getSettings() {
+
+    List<Setting<?>> settings = Arrays.asList(
+            new Setting<>(
+                    "plugin.dynamic_word_delimiter.protected_words_index",
+                    "protected_words",
```

Comment:
Since we provide a `fallbackSetting` here, we might not need to set the default values in WordDelimiterRunnable.java.
Remove the static variables and use the [get/1](https://github.com/elastic/elasticsearch/blob/v5.3.0/core/src/main/java/org/elasticsearch/common/settings/Settings.java#L245) version.

Commit that introduced the new way or registering settings: https://github.com/elastic/elasticsearch/commit/b22c526b34c5bba261ec79a083a41e40f6f1483d